Supplements of guideline of compilation on macOS with make and clean up some code

LykxSassinator opened the pull request:
I have tried to compile the KVRocks on MacOS, with 11.4 Big Sur, finding that there have missed some tools for compilation on the first time. So i have make the supplements in the README.md.
Unfortunately, it seems that there still exists some limitations and errs while compiling KVRocks on MacOS by make with scripts automatically, which should be manually interfered with. Therefore, currently, it's more recommended to use cmake on MacOS or compiling it on other Linux systems.

Commit:
Supplements of guideline of compilation on MacOS with make.

Commit:
Code clean-up and tiny improvements by diminshing unnecessary usages of Copy Constructors.

LykxSassinator commented:
It's my unpleasable push of two separatable commits. The second commit is used for making several tiny improvements by converting unnecessary usages of Copy Constructor with the appropriate movement ops.

Commit:
Revoke unnecessary refinement.

git-hulk commented:
cool, thanks for your contribution. Can u help to fix issues found by cpplint?

```
src/redis_db.cc:468:  Extra space for operator  ++;  [whitespace/operators] [4]
src/redis_hash.cc:178:  Add #include <utility> for move  [build/include_what_you_use] [4]
```

ShooterIT commented on src/redis_db.cc:
```diff
@@ -383,10 +386,8 @@ rocksdb::Status Database::FindKeyRangeWithPrefix(const std::string &prefix,
-  next_prefix.pop_back();
-  next_prefix.push_back(last_char);
+  auto prefix_size = prefix.size();
+  next_prefix[prefix_size - 1] ++;
```
```suggestion
  next_prefix[prefix_size - 1]++;
```

LykxSassinator replied:
It's my pleasure. Thx for the contributes of @ShooterIT .

ShooterIT commented on src/cron.cc:
```diff
@@ -29,7 +29,7 @@ Status Cron::SetScheduleTime(const std::vector<std::string> &args) {
       return Status(Status::NotOK, "time expression format error : " + s.Msg());
     }
-    new_schedulers.push_back(st);
+    new_schedulers.emplace_back(std::move(st));
```
i think it is not a good idea to call move, since `st` may be access again

ShooterIT replied:
actually, I also did that in parsing requests, but i find c++ standard doesn't recommend that, i just think we don't call that frequently.

ShooterIT replied:
And i find we don't define its `move constructor`, so this change doesn't take any improvements. And all members of schedule just are `int` type.

LykxSassinator replied:
Yep, at this point, std::move was not a recommendable operation for the user-defined `st` object. I would revoke it here.

LykxSassinator committed:
Apply suggestions from code review

As recommended, apply the suggestions from code review.

Co-authored-by: Wang Yuan <wangyuancode@163.com>

LykxSassinator committed:
Revoke unrecommended std::move op.

Here, I reovke unnecessary and unrecommend std::move operation on the `st` object.

LykxSassinator reviewed (commented):
All codes have been reviewed and checked again.
And It's my pleasure to make contributions to KVRocks.

ShooterIT commented on README.md:
```diff
@@ -47,7 +47,7 @@ sudo yum install -y epel-release && sudo yum install -y git gcc gcc-c++ make sna
 
 # MACOSX
-brew install snappy googletest
+brew install autoconf automake libtool snappy googletest
```
do we need to brew install gcc? I find we install gcc in CI for macOS.
https://github.com/KvrocksLabs/kvrocks/blob/unstable/.github/workflows/kvrocks.yaml#L60

LykxSassinator replied:
The original version of GCC in MacOS is `4.2.1`. Although the compilation requirement requires a higher version(gcc >= 4.8), the compilation works around in fact. So, should we recommend the MacOS users upgrade the GCC version?

ShooterIT replied:
it seems that gcc is clang on macOS platform, it is fine for compilation.

ShooterIT commented:
cpplint error https://github.com/KvrocksLabs/kvrocks/runs/3101400846#step:4:721 please add necessary header files.

Commit:
Fix the cpplint errors while checking the code-style on Ubuntu.

LykxSassinator reviewed (commented):
Done.

LykxSassinator reviewed (commented):
The error checked by cpplint has been fixed.

git-hulk reviewed (approved):
LGTM
